.github/actions/emit-metrics/action.yml

```diff
@@ -0,0 +1,74 @@
+name: Emit Metrics
+description: Emit metrics from another workflow/action using the kat tool
+inputs:
+  namespace:
+    description: The CloudWatch namespace in which to emit metrics
+    required: true
+  dimensions:
+    description: |-
+      The dimensions to include with emitted metrics, as a collection of `name=value` pairs. Multiple dimensions are
+      delimited by newlines (`\n`) and whitespace is trimmed from before/after each `name=value` pair. When passing
+      multiple dimensions, make sure to use YAML's block literal syntax (`|` or `|-`).
+
+      For example:
+      ```
+      dimensions: |
+        Artifact=foo-artifact
+        Platform=Kotlin/JVM
+        Variant=External
+      ```
+    required: true
+  metrics:
+    description: |-
+      The metrics to emit into the given namespace and using the specified dimensions. Individual metrics are written in
+      the form of `<name>:<value>[:<unit>]` where:
+      * <name> is the metric name and may include spaces but not colons (`:`)
+      * <value> is the numeric value of the metric
+      * <unit> is the CloudWatch unit name (if omitted, `None` is assumed)
+
+      Multiple metrics are delimited by newlines (`\n`) and whitespace is trimmed from before/after each metric
+      specification. When passing multiple metrics, make sure to use YAML's block literal syntax (`|` or `|-`).
+
+      For example:
+      ```
+      metrics: |
+        BuildTime:4.532:Seconds
+        BuildSucceeded:1:Count
+      ```
+    required: true
```

Comment:
Should dimensions be required? They are not marked as `.required()` in our kat tool's clikt configuration

Reply:
Oh whoops. Yes, they should be required. I cannot think of any instance in which we would not contextualize our metrics with _at least one_ dimension. I'll update the **kat** tool accordingly.